### PR TITLE
Move all modals to DenaliModal

### LINKED ISSUE #1196: Move all modals to denali

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #1196 

Currently have issues with navi-notifications.

## Screenshots
Screen Shot 2021-01-15 at 1 38 59 PM
Screen Shot 2021-01-15 at 1 38 45 PM
Screen Shot 2021-01-15 at 1 37 55 PM
Screen Shot 2021-01-15 at 1 37 44 PM
Screen Shot 2021-01-15 at 1 41 44 PM

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/templates/components/common-actions/schedule.hbs [hunk 1]
```diff
@@ -9,105 +9,73 @@
...
             {{else}}
-              <button
-                class="schedule-modal__save-btn btn btn-primary"
+              <DenaliButton
+                class="schedule-modal__save-btn m-5"
                 disabled={{this.disableSave}}
-                type="button"
                 {{on "click" (queue (toggle "isSaving" this) this.doSave)}}
               >
                 {{#if this.isSaving}}
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/common-actions/schedule.hbs:81]: This portion does not show up nicely inside the button

File: packages/reports/addon/templates/components/common-actions/schedule.hbs [hunk 2]
```diff
@@ -129,13 +97,11 @@
                 {{else}}
                   Close
                 {{/if}}
-              </button>
+              </DenaliButton>
             {{/if}}
-          </div>
-          <div class="button-group">
             {{#unless this.localDeliveryRule.isNew}}
               <CommonActions::Delete
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/common-actions/schedule.hbs:103]: This doesn't show up like a button anymore

File: packages/reports/addon/templates/components/common-actions/delete.hbs [hunk 3]
```diff
@@ -3,28 +3,28 @@
...
         disabled={{this.isDeleting}}
         {{on "click" (pipe (toggle "isDeleting" this) (fn @deleteAction @model) this.closeModal)}}
       >
         Confirm
-      </button>
-      <button type="button" class="btn btn-secondary" {{on "click" (toggle "showModal" this)}} disabled={{this.isDeleting}}>Cancel</button>
-    </div>
-  </NaviModal>
+      </DenaliButton>
+      <DenaliButton type="button m-5" class="button is-outline btn-secondary m-5" {{on "click" (toggle "showModal" this)}} disabled={{this.isDeleting}}>Cancel</DenaliButton>
```
Comment (kevinhinterlong) [inline packages/reports/addon/templates/components/common-actions/delete.hbs:27]:
```suggestion
      <DenaliButton type="button m-5" class="button is-outline m-5" {{on "click" (toggle "showModal" this)}} disabled={{this.isDeleting}}>Cancel</DenaliButton>
```

**Review by kevinhinterlong — COMMENTED** [review]
Some small comments, but some bigger things to do:
- Remove h1/h5, because it makes those modals look bolded
- `class="button"` is not necessary for `<DenaliButton />` and you should be using `@style="outline"` instead of manually adding the `'is-outline'` class

File: packages/reports/addon/templates/components/common-actions/schedule.hbs (hunk 2 above)
Comment (holderb5) [reply packages/reports/addon/templates/components/common-actions/schedule.hbs:103]: I'm aware of this, the issue is that updating the styles of this button effects all of the other buttons that spawns the delete modal, and I didnt know if I should change that yet considering the others havent been changed per Jon's request.

File: packages/reports/addon/templates/components/common-actions/schedule.hbs (hunk 1 above)
Comment (holderb5) [reply packages/reports/addon/templates/components/common-actions/schedule.hbs:81]: Think this is due to it using a NaviLoader, should I try and fix that in this pr, or just wait until my navi loader pr which should be later this week?

File: packages/dashboards/tests/acceptance/dashboards-test.js [hunk 4]
```diff
@@ -246,15 +244,15 @@ module('Acceptance | Dashboards', function(hooks) {
...
+      .dom('.modal .button')
       .hasAttribute(
         'href',
         `/dashboards/1/widgets/new`,
         'Create new assigns the new widget route to the primary button'
       );
 
     await selectChoose('.report-select', 'Report 12');
-    await click('.add-widget-modal .btn');
+    await click('. .button');
```
Comment (kevinhinterlong) [inline packages/dashboards/tests/acceptance/dashboards-test.js:255]: ?
```suggestion
    await click('.modal .button');
```

Comment (holderb5) [reply packages/dashboards/tests/acceptance/dashboards-test.js:255]: Sorry this is resolved in local I've been fixing the dashboard tests and I pushed to fix the lint when some stuff was a lil messed up

File: packages/reports/app/styles/navi-reports/components/navi-email-tag.less [hunk 5]
```diff
@@ -3,6 +3,8 @@
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
 .navi-email-tag {
+  background-color: 0x333570f4;
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/app/styles/navi-reports/components/navi-email-tag.less:6]: So this is the same color as the denali-tag color. Also I'm pretty sure scss hasnt been set up properly for reports yet.

**Pull request closed without merging**